src/module/account/settings/SettingFormGroup.tsx

```diff
@@ -0,0 +1,55 @@
+'use client';
+
+import React from 'react';
+import { Controller, useFormContext } from 'react-hook-form';
+import FormInput from './_component/FormInput/FormInput';
+
+export const SettingFormGroup = () => {
```

Comment:
Why break this out?

Reply:
nothing special

Reply:
what do you mean nothing special? hahah.

If we not reusing this then maybe it can be part of the actual setting form? or if it's for organisation purpose, I reckon we can have it colocate with the actual form?

Reply:
Sorry. Misunderstand break this out. To make SettingFormGroup a component is in alignment with Damon's example of the login page in which all the Controllers(form group in the original file) are extracted to become a component called FormGroup.

Reply:
I extracted a component for the form group because there were so many wrappers on the controller, which made the code look messy.  You built these reusable components to make it much better,  I think you just put these codes back on the same page in a function should be ok.  

what do you think @SeijaHuang  @NorrisWu0 ?

Reply:
> put these codes back on the same page in a function

wdym by this?

I guess my idea is 

```tsx

export function SettingForm() {
  const { ..... , control} = useForm(...)
  return (
    <form>
      ...
      <SettingFormGroup control={control}/>
      ...
    </for>
}

function SettingFormGroup({ control }) {
  return (
    <inputs....../>
  )
}
```

oh and, why do we use form context here, could we just pass the `control` as a props? @SeijaHuang

Reply:
sure, I will refactor here, merging the form group component into the layout. I think that will be more straightforward and will not use form context anymore.